### PR TITLE
PHP: replace php/php-dev to php7.2/php7.2-dev and delete php5 tests

### PR DESCRIPTION
**Pull request by HannahShiSFB**
This PR is part of deprecate php5. Working towards #23645

### CODE REVIEW

File: tools/run_tests/run_tests_matrix.py [hunk 1]
```diff
@@ -180,15 +180,6 @@ def _create_test_jobs(extra_args=[], inner_jobs=_DEFAULT_INNER_JOBS):
                                 ['--report_multi_target'],
                                 inner_jobs=inner_jobs)
 
-    # supported on linux only
```
Comment (stanley-cheung) [inline tools/run_tests/run_tests_matrix.py:183]: Maybe we want to keep this? This is on `php7`.

**Review by stanley-cheung — APPROVED** [review]
LGTM.

File: tools/run_tests/run_tests_matrix.py (hunk 1 above)
Comment (HannahShiSFB) [reply tools/run_tests/run_tests_matrix.py:183]: It's included later at https://github.com/HannahShiSFB/grpc/blob/60b5d06ec1c069e96dc7163aad3412754716ff56/tools/run_tests/run_tests_matrix.py#L233

**Comment by veblush** [comment]
This is just rolled back and please update distribution test with PHP7 when trying to make a fix-forward.

**Pull request merged**